Bump dependency versions

TedBrookings opened the pull request:
My newish version of rustc couldn't build the older rust_htslib dependency. So this does two things:
1. Bump dependency version to the latest.
2. Change one deprecated function invocation to the new name.

With these fixes I was able to build.

Commit:
Bump dependency versions

Commit:
Fix deprecated method to new method name

TedBrookings commented:
Obviously this is an issue I should have noticed. I got so excited that I could use the library I didn't check the library tests.
I'm a little confused about how this is erroring now, I'll look into it when I get some time.

sstadick commented:
I'll try to take a look later this week if you haven't gotten to it by then! Thanks for making the PR!

Commit:
Manually unset unmapped flag

Commit:
Change MQ from Char to U8

TedBrookings commented:
I added two commits.

1. Manually unset unmapped flag when the ReadPairSpec calls for it. It seems that the upgrade caused reads to default to unmapped. now. This fixes the test failures.
2. While searching for the problem, I noticed `MQ` was set to type `Char` and I thought `U8` might be more appropriate and it might cause problems. Feel free to reject this change, it's not the source of the problem. I moved it to the end to make it easy to pop off.

sstadick commented:
This looks good to me! I like the change to U8 as well, thanks for fixing that up.

TedBrookings commented:
Awesome, thanks for getting the release out!
